### PR TITLE
Ensure filtered list does not change after edits

### LINKED ISSUE #110: Editing person after filter results in refresh
1. Perform a filter function with `find` or `absentees` command
2. Try to edit that person with `edit` `unmark` `mark` `grade` `deletegrade`, 
3. filtered list resets

If the user is focused on that filtered entry, entry updates performed should be reflected on the filtered list.

### PR DESCRIPTION
**Pull request by Oceankoh**
Fix #110

### CODE REVIEW

**Comment by Oceankoh** [comment]
With this change the app now does the following
scenario 1 
1. `find Alex`
2. `edit 1 n/Alex Wong`
3. listed people still shows only one person

scenario 2 (should this be correct behavior?)
1. `find Alex`
2. `edit 1 n/Wong`
3. listed people still shows no one (since there is no more alex)

**Comment by izruff** [comment]
> scenario 2 (should this be correct behavior?)
> 
> 1. `find Alex`
> 2. `edit 1 n/Wong`
> 3. listed people still shows no one (since there is no more alex)

It makes sense, and implementing the alternative way (new list should show Wong) would be technically challenging. Let's keep it this way for now.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -171,6 +171,8 @@ This app is designed for Teaching Assistants (TAs) who are experienced with usin
 - Extraneous parameters for commands that do not take in parameters (such as `help`, `list`, `exit` and `clear`) will be ignored.<br>
   e.g. if the command specifies `help 123`, it will be interpreted as `help`.
 
+- Dates supplied by the user are assumed to be in Anno Domini or the Common Era. In other words, the user will not be able to enter a date before the year 0001 AD. 
+
```
Comment (izruff) [inline docs/UserGuide.md:175]: Wrong PR but ok I guess

File: src/main/java/seedu/address/logic/commands/AddGradeCommand.java [hunk 2]
```diff
@@ -64,7 +63,7 @@ public CommandResult execute(Model model) throws CommandException {
         try {
             Person updatedPerson = personToEdit.addGrade(this.toAdd);
             model.setPerson(personToEdit, updatedPerson);
-            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            model.updateFilteredPersonList(model.getCurrentPersonFilter());
```
Comment (izruff) [inline src/main/java/seedu/address/logic/commands/AddGradeCommand.java:66]: Is there a reason for this? It's essentially reassigning the same predicate.

I tried removing this line from `EditCommand` and the behavior seems identical.

**Review by izruff — COMMENTED** [review]
Seems fine to me but I have a question

File: src/main/java/seedu/address/logic/commands/AddGradeCommand.java (hunk 2 above)
Comment (Oceankoh) [reply src/main/java/seedu/address/logic/commands/AddGradeCommand.java]: You're right. I was under the impression that we had to call update to display any changes made to the list, but this is not the case. I've removed the unnecessary commands!

File: docs/UserGuide.md (hunk 1 above)
Comment (Oceankoh) [reply docs/UserGuide.md:175]: oh right oops

**Review by izruff — APPROVED** [review]
LGTM

**Review by juancarlovieri — APPROVED** [review]
lgtm

**Pull request merged**